synapse/appservice/__init__.py

```diff
@@ -124,29 +125,23 @@ def _check_namespaces(self, namespaces):
                     raise ValueError(
                         "Expected bool for 'exclusive' in ns '%s'" % ns
                     )
-                if not isinstance(regex_obj.get("regex"), basestring):
+                regex = regex_obj.get("regex")
+                if isinstance(regex, basestring):
+                    regex_obj["regex"] = re.compile(regex)
```

Comment:
Err, like this?

```python
regex_obj["regex"] = re.compile(regex)  # Pre-compile regex
```

Isn't that commenting on what a line does? I don't see how it helps understanding?